Sources/StreamChatUI/UIConfig+SwiftUI.swift

```diff
@@ -0,0 +1,31 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import SwiftUI
+
+@available(iOS 13.0, *)
+@dynamicMemberLookup
+/// `UIConfig` represented as `ObservableObject` class for SwiftUI requirements.
+public class SwiftUIConfig: ObservableObject {
+    public subscript<T>(dynamicMember keyPath: WritableKeyPath<UIConfig, T>) -> T {
+        get { StreamChatUI.UIConfig.default[keyPath: keyPath] }
+        set { StreamChatUI.UIConfig.default[keyPath: keyPath] = newValue }
+    }
+}
+
+@available(iOS 13.0, *)
+/// Modifier for setting `SwiftUIConfig` environment object.
+private struct SwiftUIEnvironment: ViewModifier {
+    public func body(content: Content) -> some View {
+        content.environmentObject(SwiftUIConfig())
+    }
+}
+
+@available(iOS 13.0, *)
+public extension View {
+    /// Sets up UI environment.
+    func setUpStreamChatUIEnvironment() -> some View {
+        self.modifier(SwiftUIEnvironment())
+    }
```

Comment:
Which `SwiftUIConfig` is used needs to be customizable, this always uses `_UIConfig<NopExtraData>.default` which should ideally be the default value.

Maybe we should rename this to `setStreamChatUIConfig()`?

Reply:
Adjusted in 20235c54ffa29f90506859e1123c29b03957f826.